gemini/evaluation/evaluating_prompts_at_scale_with_gemini_batch_prediction_api.ipynb

```diff
@@ -0,0 +1,1011 @@
+    "        FROM `{files_table}` f\n",
+    "        JOIN `{results_view}` e\n",
+    "          ON f.path = e.gcs_uri\n",
+    "      )\n",
+    "\n",
+    "      SELECT\n",
+    "        evaluation_id,\n",
+    "        evaluation_ts,\n",
+    "        prompt_text,\n",
+    "        path,\n",
+    "        label,\n",
+    "        correct\n",
+    "      FROM t1\"\"\"\n",
+    "\n",
+    "    # Make an API request to create the view.\n",
+    "    view = bq_client.create_table(view, exists_ok=False)\n",
+    "    print(f\"Created {view.table_type}: {str(view.reference)}\")\n",
+    "\n",
+    "\n",
+    "create_evaluation_view(\n",
+    "    evaluation_view=EVALUATION_VIEW, files_table=FILES_TABLE, results_view=RESULTS_VIEW\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "Xn2t64pkbQHq"
+   },
+   "source": [
+    "## Define prompt and execute it via Gemini Batch Prediction API"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "Xbq75NqxRXBO"
+   },
+   "source": [
+    "### Define the prompt"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 83,
+   "metadata": {
+    "id": "bk-EpTp0uJR4"
+   },
+   "outputs": [],
+   "source": [
+    "prompt = \"\"\"\\\n",
+    "- Classify the sport from the image below in one of the following categories:\n",
+    "* baseball\n",
+    "* basketball\n",
+    "* tennis\n",
+    "* volleyball\n",
+    "\n",
+    "- Provide an answer in JSON format. 3. Example response:\n",
+    "'{\"sport\": \"baseball\"}'\n",
+    "\n",
+    "- Image:\n",
+    "\"\"\""
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "XCGQhmmSRbgc"
+   },
+   "source": [
+    "### Classify one image using the Python SDK"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "colab": {
+     "base_uri": "https://localhost:8080/"
+    },
+    "id": "Kchsx7bdrGgf",
+    "outputId": "45411834-7bee-4d00-8e60-6541df113b2d"
+   },
+   "outputs": [],
+   "source": [
+    "def download_blob_into_memory(bucket_name, blob_name):\n",
+    "    \"\"\"Downloads a blob from GCS into memory.\"\"\"\n",
+    "    bucket = storage_client.bucket(bucket_name)\n",
+    "\n",
+    "    blob = bucket.blob(blob_name)\n",
+    "    contents = blob.download_as_bytes()\n",
+    "    return contents\n",
+    "\n",
+    "\n",
+    "def classify_image(model_id, prompt, bucket_name, blob_name):\n",
+    "    \"\"\"Classifies an image.\"\"\"\n",
+    "    model = GenerativeModel(model_id)\n",
+    "    contents = download_blob_into_memory(bucket_name, blob_name)\n",
+    "    encoded_image = base64.b64encode(contents).decode(\"utf-8\")\n",
+    "    image_content = Part.from_data(\n",
+    "        data=base64.b64decode(encoded_image), mime_type=\"image/jpeg\"\n",
+    "    )\n",
+    "    contents = [prompt, image_content]\n",
+    "    response = model.generate_content(contents)\n",
+    "    return response\n",
+    "\n",
```

Comment:
You don't need to download the image locally. It's preferred to send the GCS uri directly. Example:

https://cloud.google.com/vertex-ai/generative-ai/docs/multimodal/image-understanding#gemini-send-multimodal-samples-image-python

Reply:
Batch prediction doesn't support the fileData field for Gemini. See https://cloud.google.com/vertex-ai/generative-ai/docs/multimodal/batch-prediction-gemini#prepare_your_inputs